lib/sinon/util/fake_xdomain_request.js

```diff
@@ -7,8 +7,24 @@
 /**
  * Fake XDomainRequest object
  */
+
+/**
+ * Returns the global to prevent assigning values to 'this' when this is undefined.
+ * This can occur when files are interpreted by node in strict mode.
+ * @private
+ */
+function getGlobal() {
+    "use strict";
+
+    return typeof window !== "undefined" ? window : global;
```

Comment:
You are absolutely correct on that. I was wondering this myself yesterday. I have made a small test now that I will submit as a PR that will simply test if sinon works in a web worker.